proposals/0000-specify-dump-json.md

```diff
@@ -0,0 +1,155 @@
+# RFC - Specify GHC's JSON Diagnostic Dump
+
+## Abstract
+
+GHC is currently undergoing a long term project to [move to a more structured error representation](https://gitlab.haskell.org/ghc/ghc/-/issues/18516). 
+In harmony with this motion, there is a [desire for a JSON dump of GHC's diagnostic messages](https://gitlab.haskell.org/ghc/ghc/-/issues/19278).
+Such an implementation would facilitate easier consumption for downstream consumers of GHC's diagnostic messages (e.g. IDE developers). 
+To best enforce a standardized JSON output, a [JSON schema](https://json-schema.org/) can be used.
+
+The purpose of this RFC is to open up the discussion to the broader Haskell community about:
+
+- the desire for such a feature
+- the requested fields present in such a JSON output
+- opinions on the draft JSON schema (or the usage of JSON schemas at all)
+- any other general feedback 
+
+## Background
+
+In the prior implementation of GHC, errors were passed around as `SDoc`s (for Structured Documents).
+These `SDoc`s are simply a text encoding of the errors with some formatting features, such as nice alignment.
+By transforming the errors into `SDoc`s at the moment of creation, a lot of type-level information is lost,
+causing a headache for GHC developers and GHC API users.
+To improve on this problem, the [journey to represent errors as structured values has begun](https://gitlab.haskell.org/ghc/ghc/-/wikis/Errors-as-(structured)-values ). 
+In summary, this means passing errors around as a specific type which contains the relevant information separated into different fields,
+rather than combining it all into one single `SDoc`.
+
+Another change that has been made is the introduction of [error codes](https://gitlab.haskell.org/ghc/ghc/-/issues/21684) to uniquely identify each possible error output from GHC. 
+
+These features together serve to make error consumption a breeze for GHC API users, but what about those that would like to consume errors without utilizing GHC's API?
```

Comment:
I commented on the issue, but just to repeat here: HLS uses the GHC API, so it's not one of the parties who would benefit from this.